doc/guides/networking/net-stack-architecture.rst

```diff
@@ -0,0 +1,137 @@
+.. _network_stack_architecture:
+
+Network Stack Architecture
+##########################
+
+High level overview of the network stack
+****************************************
+
+.. figure:: zephyr_netstack_overview.svg
+    :align: left
+    :alt: Overview of the network stack architecture
+    :figclass: align-center
+
+    Network stack overview
+
+*The network stack is layered and it consists of the following parts:*
+
+* **Network Application.** The network application uses
+  :ref:`BSD socket API <bsd_sockets_interface>` to create a network connection,
+  send or receive data, and close a connection. The application can also use
+  the :ref:`network management API <net_mgmt_interface>` to set network
+  related parameters such as taking network interface down, starting a scan
+  (when applicable), setting IP address to a network interface, etc.
+
+* **Network Protocols.** This provides implementations for
```

Comment:
Please split this in more bullet points:

* Application-level network protocols.
* BSD Socket API
* Network protocols

Each could have a sentence or two of description, and this describes layering more explicitly and understandably.

Reply:
I created bullet points but left out the BSD socket API as that is not a protocol.

Reply:
> I created bullet points but left out the BSD socket API as that is not a protocol.

It's not a protocol, it's *layer*. These bullet points describe layers.

Reply:
We are talking about protocols here in this bullet, not layers. I will mention socket API in core network protocol bullet instead. Will send a new version shortly so you see what I mean.

Reply:
> We are talking about protocols here in this bullet, not layers.

And what I suggested in the original comment, that some layers are not described, and instead of 1 bullet point, have 3. (Colloquially, "split in 3").

> I will mention socket API in core network protocol bullet instead. 

It's a separate layer on its own. Both paradigmatically and code-wise.